internal/utils/nodeGroup.go

```diff
@@ -23,90 +29,144 @@ func CalculateNodeGroup(ctx context.Context, nodes v1.NodeList, config danav1alp
 				ResourceMultiplier = resourceGroup.ResourceMultiplier
 			}
 		}
-
 	}
 	nodeGroupReources := v1.ResourceList{}
 	for _, node := range nodes.Items {
-		resources := MultiplyResourceList(node.Status.Allocatable, ResourceMultiplier)
+		resources := multiplyResourceList(node.Status.Allocatable, ResourceMultiplier)
 		for resourceName, resourceQuantity := range resources {
-			AddResourcesToList(&nodeGroupReources, resourceQuantity, string(resourceName))
+			addResourcesToList(&nodeGroupReources, resourceQuantity, string(resourceName))
 		}
 	}
 
-	return nodeGroupReources
+	return filterUncontrolledResources(nodeGroupReources, config.Spec.ControlledResources)
 }
 
-func CaculateGroupReservedResources(reserved []danav1alpha1.ReservedResources, group string, hoursLimit int) v1.ResourceList {
-	resources := v1.ResourceList{}
-	for _, resource := range reserved {
-		if resource.NodeGroup == group && HoursPassedSinceDate(resource.Timestamp) < hoursLimit {
-			for resourceName, resourceQuantity := range resources {
-				AddResourcesToList(&resources, resourceQuantity, string(resourceName))
-			}
+// getReservedResourcesByGroup retrieves the reserved resources for a specific node group from the NodeQuotaConfig.
+// It takes the node group name and the NodeQuotaConfig.
+// It returns the ReservedResources object (danav1alpha1.ReservedResources) for the node group.
+func getReservedResourcesByGroup(group string, config danav1alpha1.NodeQuotaConfig) danav1alpha1.ReservedResources {
+	if !doesReservedResourceExist(config, group) {
+		return danav1alpha1.ReservedResources{}
+	}
+	for _, resource := range config.Status.ReservedResources {
+		if resource.NodeGroup == group {
+			return resource
 		}
 	}
-	return resources
+	return danav1alpha1.ReservedResources{}
 }
 
-func DeleteExpiredReservedResources(config *danav1alpha1.NodeQuotaConfig) {
+// DeleteExpiredReservedResources removes the expired reserved resources from the NodeQuotaConfig.
+// It takes the NodeQuotaConfig to modify and a logger for logging informational messages.
+func DeleteExpiredReservedResources(config *danav1alpha1.NodeQuotaConfig, logger logr.Logger) {
 	newReservedResources := []danav1alpha1.ReservedResources{}
 	for _, resources := range config.Status.ReservedResources {
-		if HoursPassedSinceDate(resources.Timestamp) < int(config.Spec.ReservedHoursTolive) {
+		if hoursPassedSinceDate(resources.Timestamp) < int(config.Spec.ReservedHoursTolive) {
```

Comment:
This condition is often used and should be a simple function with an indicative name instead